diff --git a/pass/online.cpp b/pass/online.cpp
index 2c29a59..18f6ad3 100644
--- a/pass/online.cpp
+++ b/pass/online.cpp
@@ -34,7 +34,7 @@
 #include "llvm/IR/Module.h"
 #include "llvm/Pass.h"
 #include "llvm/Passes/PassBuilder.h"
-#include "llvm/Plugins/PassPlugin.h"
+#include "llvm/Passes/PassPlugin.h"
 #include "llvm/Support/Casting.h"
 #include "llvm/Support/CommandLine.h"
 #include "llvm/Support/Error.h"
